pkg/gadgets/trace/dns/tracer/tracer.go

```diff
@@ -332,30 +132,80 @@ func (t *Tracer) install() error {
 	return nil
 }
 
-func (t *Tracer) run(ctx context.Context, logger logger.Logger, dnsTimeout time.Duration) error {
-	spec, err := loadDns()
+func (t *Tracer) parseDNSPacket(rawSample []byte, netns uint64) (*types.Event, error) {
+	// The sample received is a concatenation of the dnsEventT structure and the packet bytes.
+	bpfEvent := (*dnsEventT)(unsafe.Pointer(&rawSample[0]))
+	packetBytes := rawSample[unsafe.Sizeof(*bpfEvent):]
+
+	// The DNS layer starts after the ethernet, IP and UDP headers.
+	DNS_OFF := 14 + 20 + 8
```

Comment:
Some comments for this line?

- dnsOffset would me more go-style?
- Isn't there defines like `ETH_HLEN` in gopacket pkg?
- Is it safe to consider the IP header with a fixed size (20)?

Reply:
It's all removed in next commit, I didn't want to expend a lot of time in something that will be removed one commit later. Again, I'll happy to squash both commits if preferred.